### PR TITLE
Adding KT tutorial text for tutorials 4-6

### PR DESCRIPTION
**Pull request by SHeimlich**

### CODE REVIEW

File: particles/Tutorial/Kotlin/README.md [hunk 1]
```diff
@@ -212,3 +212,367 @@ arcs_kt_binary(
 
 And there you have it! The mystery of root solved, and a basic understanding of slots. Slots are a large part of the power of Arcs to hide user data, so we'll be using them a lot going forward. So don't worry if you don't fully understand them yet, there will plenty more examples to come!
 
+# Getting a Grip on Handles
```
Comment (jasonwyatt) [inline particles/Tutorial/Kotlin/README.md]: Nice pun

File: particles/Tutorial/Kotlin/README.md [hunk 2]
```diff
@@ -212,3 +212,367 @@ arcs_kt_binary(
...
+
+```kotlin
+class GetPersonParticle : Particle() {
+    // We start by setting the type of the handle. Note that this is a Singleton handle, as we will only have one person.
+    // the GetPerson_Person() entity class is autogenerated from the Bazel build rules. 
+    private val person = Singleton { GetPerson_Person() }
+
+    override fun getTemplate(slotName: String) = """
+        <input placeholder="Enter your name" spellcheck="false" on-change="onNameInputChange">
+        <div slotid="greetingSlot"></div>""".trimIndent()
```
Comment (jasonwyatt) [inline particles/Tutorial/Kotlin/README.md]: with `trimIndent` you should be able to put the closing `"""` on the next line. That might help a little with readability.

File: particles/Tutorial/Kotlin/README.md [hunk 3]
```diff
@@ -212,3 +212,367 @@ arcs_kt_binary(
...
+class CollectionsParticle : Particle() {
+    // Note this is a Collection handle instead of a Singleton.
+   private val people = Collection { CollectionsParticle_InputData() }
+
+   init {
+       registerHandle("inputData", people)
+   }
+
+    override fun populateModel(slotName: String, model: Map<String, Any?>): Map<String, Any?> {
+        // We begin by generating the list of models that should fill the template. Our template has name ang age so we
```
Comment (jasonwyatt) [inline particles/Tutorial/Kotlin/README.md]: Might be good to limit the column width of these examples to something like 100 or 120 chars.

File: particles/Tutorial/Kotlin/README.md [hunk 4]
```diff
@@ -212,3 +212,367 @@ arcs_kt_binary(
...
+)
+
+arcs_kt_binary(
+    name = "Collections",
+    srcs = ["Collections.kt"],
+    deps = [":collections_schemas"],
+)
+```
+
+When you execute this recipe, you should see everyone being greeted. If you added or remove an entity from the `PeopleData` store, the number of people greeted will change accordingly thanks to the template interpolation.
```
Comment (jasonwyatt) [inline particles/Tutorial/Kotlin/README.md]: "If you ***add*** or remove an entity..."

File: particles/Tutorial/Kotlin/README.md [hunk 5]
```diff
@@ -212,3 +212,367 @@ arcs_kt_binary(
...
+
+/**
+ * Sample Kotlin-WASM Particle to use a JSON store.
+ */
+class JsonStoreParticle : Particle() {
+
+    private val res = Singleton { JsonStoreParticle_InputData() }
+    init {
+        registerHandle("inputData", res)
+    }
```
Comment (jasonwyatt) [inline particles/Tutorial/Kotlin/README.md]: The whitespace here is a little funky

File: particles/Tutorial/Kotlin/README.md (hunk 1 above)
Comment (SHeimlich) [reply particles/Tutorial/Kotlin/README.md]: Thanks :)

File: particles/Tutorial/Kotlin/README.md [hunk 6]
```diff
@@ -212,3 +212,371 @@ arcs_kt_binary(
...
+
+    override fun onHandleUpdate(handle: Handle) {
+        this.renderOutput()
+    }
+
+    override fun populateModel(slotName: String, model: Map<String, Any?>): Map<String, Any?> {
+        val n = person.get()?.name ?: "Human"
+        return model + mapOf(
+            "name" to n
```
Comment (piotrswigon) [inline particles/Tutorial/Kotlin/README.md]: inline `n` here maybe?

**Pull request merged**